pep-0496-environment-markers.rst

```diff
@@ -48,14 +50,32 @@ Environment Markers attempt to provide more flexibility in a list of
 requirements by allowing the developer to list requirements that are
 specific to a particular environment.
 
+Alternatives considered
+=======================
+
+For a while, OpenStack used seperate requirements files for different
+environments - eg, ``requirements-py2.txt`` and
+``requirements-py3.txt``. However, in most cases the two files were
+near duplicates, with only one or two lines different; but maintaining
+two seperate files was more than twice as much work, as it required
+manual checking to keep the files in sync.
```

Comment:
That wasn't the problem OpenStack had. The problem was building universal wheels was impossible. Also, markers predate OpenStack's efforts in this space, this PEP is updating the definition to be more useful based on lessons learnt - e.g. dealing with 2.7.10.